### PR TITLE
Add bun support

### PR DESCRIPTION
**Pull request by megatux**
Related to issue #10

### CODE REVIEW

File: lib/generators/htmx/install_generator.rb [hunk 1]
```diff
@@ -40,38 +46,34 @@ def manifest(javascript_dir)
...
-        manifest = manifest('app/assets/javascripts')
+      def setup_bun
+        run 'bun add htmx.org'
 
-        if manifest.exist?
-          append_file manifest, "\n#{SPROCKETS_SETUP}"
-        else
-          create_file manifest, SPROCKETS_SETUP
-        end
+        add_to_manifest(manifest('app/javascript'), IMPORTMAP_SETUP)
```
Comment (blacksam07) [inline lib/generators/htmx/install_generator.rb:60]: Should be `BUN_SETUP` or we will use the same as importmap?

File: lib/generators/htmx/install_generator.rb [hunk 2]
```diff
@@ -11,7 +11,9 @@ class InstallGenerator < ::Rails::Generators::Base
 
       # Setup HTMX
       def setup
-        if importmap?
+        if bun_configured?
```
Comment (blacksam07) [inline lib/generators/htmx/install_generator.rb:14]: To follow up the method naming style, can we rename to `bun?`

**Review by blacksam07 — CHANGES_REQUESTED** [review]
Can we update the `desc` method in the line 10 of `install_generator.rb` file and include the reference to bun

File: lib/generators/htmx/install_generator.rb [hunk 3]
```diff
@@ -40,38 +46,34 @@ def manifest(javascript_dir)
...
         else
-          create_file manifest, IMPORTMAP_SETUP
+          create_file manifest, text
         end
       end
 
-      def setup_sprockets
-        manifest = manifest('app/assets/javascripts')
+      def setup_bun
+        run 'bun add htmx.org'
```
Comment (JulianPasquale) [inline lib/generators/htmx/install_generator.rb:58]: I think this is installing the latest version of htmx. I'm wondering if we should fix the version here - and probably apply that to the rest of the JS bundling methods as well

Comment (JulianPasquale) [reply lib/generators/htmx/install_generator.rb:58]:
```suggestion
        run "bun add htmx.org@#{Htmx::Rails::HTMX_VERSION}"
```

File: lib/generators/htmx/install_generator.rb (hunk 1 above)
Comment (megatux) [reply lib/generators/htmx/install_generator.rb]: I'll add a new constant

**Review by JulianPasquale — APPROVED** [review]
Looks good!

CI is failing but I don't think that is related to this PR

**Review by blacksam07 — APPROVED** [review]
LGTM, thanks

**Comment by megatux** [comment]
> Looks good!
> 
> CI is failing but I don't think that is related to this PR

Right, seems related to this [Truffleruby issue](https://github.com/oracle/truffleruby/issues/3328)

**Pull request merged**